FetchMigration/index_configuration_tool/index_operations.py

```diff
@@ -24,14 +25,21 @@ def fetch_all_indices(endpoint: str, optional_auth: Optional[tuple] = None, veri
     return result
 
 
-def create_indices(indices_data: dict, endpoint: str, auth_tuple: Optional[tuple]):
+def create_indices(indices_data: dict, endpoint: EndpointInfo):
     for index in indices_data:
-        actual_endpoint = endpoint + index
+        actual_endpoint = endpoint.url + index
         data_dict = dict()
         data_dict[SETTINGS_KEY] = indices_data[index][SETTINGS_KEY]
         data_dict[MAPPINGS_KEY] = indices_data[index][MAPPINGS_KEY]
         try:
-            resp = requests.put(actual_endpoint, auth=auth_tuple, json=data_dict)
+            resp = requests.put(actual_endpoint, auth=endpoint.auth, verify=endpoint.verify_ssl, json=data_dict)
             resp.raise_for_status()
         except requests.exceptions.RequestException as e:
-            print(f"Failed to create index [{index}] - {e!s}", file=sys.stderr)
+            raise RuntimeError(f"Failed to create index [{index}] - {e!s}")
+
+
+def doc_count(indices: set, endpoint: EndpointInfo) -> int:
```

Comment:
Should there be some try/catching in here?

Reply:
on the requests, but also casting to an int for a key that we're assumign is in a dict is always risky

Reply:
The `count` member of the [_count API](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/search-count.html) will always be an int so I think that part of it is safe.

I'll add a backlog item to improve try-catching on requests throughout the FetchMigration tool (they're all sadly lacking)